x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/OrdinalsGroupingOperator.java

```diff
@@ -448,4 +462,49 @@ public void close() {
             Releasables.close(extractor, aggregator);
         }
     }
+
+    static final class BlockOrdinalsReader {
+        private final SortedSetDocValues sortedSetDocValues;
+        private final Thread creationThread;
+
+        BlockOrdinalsReader(SortedSetDocValues sortedSetDocValues) {
+            this.sortedSetDocValues = sortedSetDocValues;
+            this.creationThread = Thread.currentThread();
+        }
+
+        IntBlock readOrdinalsAdded1(IntVector docs) throws IOException {
```

Comment:
I moved this class to OrdinalsGroupingOperator since it's only used by this class and the new behavior (add 1 to the ordinals). We can generalize it later if necessary.